app/src/main/java/de/eknoes/inofficialgolem/ArticleListFragment.java

```diff
@@ -194,64 +197,32 @@ public void putBitmap(String url, Bitmap bitmap) {
-            };
-
-            String sort = FeedReaderContract.Article.COLUMN_NAME_DATE + " DESC";
-            String limit = "0, " + PreferenceManager.getDefaultSharedPreferences(context).getInt("article_limit", 200);
-
-
-            cursor = db.query(
-                    FeedReaderContract.Article.TABLE_NAME,
-                    columns,
-                    null,
-                    null,
-                    null,
-                    null,
-                    sort,
-                    limit);
+            QueryRequest queryRequest = new QueryRequest.QueryRequestBuilder()
+                    .withSort(DBColumns.COLUMN_NAME_DATE + " DESC")
+                    .withLimit("0, " + PreferenceManager.getDefaultSharedPreferences(context).getInt("article_limit", 200))
+                    .withTableName(DBColumns.getTableName())
+                    .build();
+
+            articles = DBHelper.getArticles(queryRequest);
+
         }
 
         @Override
         public int getCount() {
-            return cursor.getCount();
+            return articles.size();
         }
 
         @Override
         public Article getItem(int position) {
-            cursor.moveToPosition(position);
-            Article a = new Article();
-            a.setId(cursor.getInt(cursor.getColumnIndexOrThrow(FeedReaderContract.Article.COLUMN_NAME_ID)));
-            a.setTitle(cursor.getString(cursor.getColumnIndexOrThrow(FeedReaderContract.Article.COLUMN_NAME_TITLE)));
-            a.setSubheadline(cursor.getString(cursor.getColumnIndexOrThrow(FeedReaderContract.Article.COLUMN_NAME_SUBHEADING)));
-            a.setTeaser(cursor.getString(cursor.getColumnIndexOrThrow(FeedReaderContract.Article.COLUMN_NAME_TEASER)));
-            a.setDate(cursor.getLong(cursor.getColumnIndexOrThrow(FeedReaderContract.Article.COLUMN_NAME_DATE)));
-            a.setImgUrl(cursor.getString(cursor.getColumnIndexOrThrow(FeedReaderContract.Article.COLUMN_NAME_IMG)));
-            a.setCommentUrl(cursor.getString(cursor.getColumnIndexOrThrow(FeedReaderContract.Article.COLUMN_NAME_COMMENTURL)));
-            a.setCommentNr(cursor.getString(cursor.getColumnIndexOrThrow(FeedReaderContract.Article.COLUMN_NAME_COMMENTNR)));
-            a.setUrl(cursor.getString(cursor.getColumnIndexOrThrow(FeedReaderContract.Article.COLUMN_NAME_URL)));
-            a.setOffline(cursor.getInt(cursor.getColumnIndexOrThrow(FeedReaderContract.Article.COLUMN_NAME_OFFLINE)) == 1);
-            a.setFulltext(cursor.getString(cursor.getColumnIndexOrThrow(FeedReaderContract.Article.COLUMN_NAME_FULLTEXT)));
-            return a;
+            return articles.get(position);
         }
 
         @Override
         public long getItemId(int position) {
-            cursor.moveToPosition(position);
-            return cursor.getLong(cursor.getColumnIndexOrThrow(FeedReaderContract.Article.COLUMN_NAME_ID));
+            return articles.get(position).getId();
         }
 
+        @RequiresApi(api = Build.VERSION_CODES.O)
```

Comment:
Warum brauchen wir das? Alleine in der Play-Version sind knapp 200 Personen, die noch unter Android 8 sind. Bekommst du das auch ohne ein API Requirement hin? :)